packages/ses/src/error/assert.js

```diff
@@ -249,34 +249,19 @@ export { loggedErrorHandler };
 // /////////////////////////////////////////////////////////////////////////////
 
 /**
- * Makes and returns an `assert` function object that shares the bookkeeping
```

Comment:
I'm not fond of moving the comments away from the code like this. The `MakeAssert` type isn't shared with anything else, is it? Is there some reason to use `@callback` and such?

Reply:
There is a reason to give it a distinct type name and move it into `types.js`. This PR also makes `makeAssert` a member of the `assert` object, which is itself typed in `types.js`. Also, it is the convention used in general for this code and increasingly across our system.

Separately is the question of whether to use `@callback` typing or `@typedef` typing for typing functions. We usually use `@callback` typing but I've occasionally found myself switching from one to the other, but I don't really understand the tradeoffs between these two choices. 

@michaelfig which way do you recommend for declaring a named function type?
@dckc if @michaelfig recommends switching to a `@typedef` it is easy enough to change later that I'm inclined to proceed merging this.

Reply:
>  This PR also makes makeAssert a member of the assert object, which is itself typed ...

OK, I see the sharing now.

I still don't see motivation for this convention of moving JSDoc away from code, but that's only tangentially related. (I suppose I owe you a draft code review checklist...)

And by all means, proceed. This comment wasn't critical. If it were, I wouldn't have approved.

Reply:
> @michaelfig which way do you recommend for declaring a named function type?

Either is fine.  It's a question of how terse you want to be.

> @dckc if @michaelfig recommends switching to a `@typedef` it is easy enough to change later that I'm inclined to proceed merging this.

No need to switch.